dawn/src/dawn/CodeGen/CXXNaive-ico/ASTStencilBody.cpp

```diff
@@ -305,7 +305,8 @@ void ASTStencilBody::visit(const std::shared_ptr<iir::FieldAccessExpr>& expr) {
         if(expr->getOffset().hasVerticalIndirection()) {
           ss_ << ","
               << "m_" << expr->getOffset().getVerticalIndirectionFieldName() << "(deref(LibTag{}, "
-              << sparseArgName_ << "), k)+" << expr->getOffset().verticalShift() << ")";
+              << sparseArgName_ << "), " << sparseIdx << ", k)+"
+              << expr->getOffset().verticalShift() << ")";
```

Comment:
Before it was working for a dense IndexField, now only for a sparse IndexField. Both cases need to be codegen'd.